deepview_profile/utils.py

```diff
@@ -1,10 +1,67 @@
 import torch 
 import logging
 import gc
+import os
+import base64
+
+from google.protobuf.json_format import MessageToDict
 
 logger = logging.getLogger(__name__)
 
 def release_memory():
     logger.debug("Emptying cache")
     gc.collect()
-    torch.cuda.empty_cache()
\ No newline at end of file
+    torch.cuda.empty_cache()
+
+def next_message_to_dict(object):
+    message = next(object)
+    return MessageToDict(message)
+
+def files_encoded_unique(operation_tree):
+    encoded_files = []
+
+    for analysis in operation_tree:
+        context_info_map = analysis['operation']['contextInfoMap']
+        if len(context_info_map) > 0:
+            filename = list(context_info_map[0]['context']['filePath']['components']).pop()
+
+            try: 
+                file_index = encoded_files.index(filename)
+            except:
+                file_path = os.path.join("", context_info_map[0]['context']['filePath']['components'])
```

Comment:
context_info_map[0]['context']['filePath']['components'] is a list, if you do os.path.join it throws an error.
File "/home/john/temp/deepview/DeepView.Profile/deepview_profile/utils.py", line 31, in files_encoded_unique
file_path = os.path.join("", context_info_map[0]['context']['filePath']['components'])
....
TypeError: join() argument must be str, bytes, or os.PathLike object, not 'list'

**This is my recommendation:**

**get file name and file path** 
**def files_encoded_unique**(operation_tree):
    encoded_files = []

    for analysis in operation_tree:
        context_info_map = analysis["operation"].get("contextInfoMap", None)
        if context_info_map is not None and len(context_info_map) > 0:
            filename = context_info_map[0]["context"]["filePath"]["components"][-1]
            file_path = "/".join(
                context_info_map[0]["context"]["filePath"]["components"]
            )
            already_in_list = next(
                (item for item in encoded_files if item["name"] == filename), None
            )
            if not already_in_list:
                encoded_file = encode_file(file_path, filename)
                encoded_files.append(encoded_file)

    return encoded_files

**then pass these two parameters to the encoder**

def encode_file(file_path, filename):
    file_dict = None
    if os.path.splitext(file_path)[1] == ".py" and filename != "entry_point.py":
        file_dict = {"name": filename, "content": ""}

        with open(file_path, "r") as f:
            file_content = f.read()
            file_dict["content"] = base64.b64encode(
                file_content.encode("utf-8")
            ).decode("utf-8")

    return file_dict

and please delete the function **files_encoded_content**, it seems is not being used

Reply:
Hi @OlegGolfer7 ,

It looks good, can you please delete line 30 in utils.py 
print(f"new file: {filename}")

I was just printing for debugging purposes.
Thanks

Reply:
Thank you. Resolved it.